docs/ref/tasks.txt

```diff
@@ -0,0 +1,318 @@
+
+.. attribute:: ResultStatus.SUCCEEDED
+
+    The :class:`Task` has finished executing successfully.
+
+.. class:: TaskResult
+
+The ``TaskResult`` stores the information about a specific execution of a
+:class:`Task`.
+
+Attributes of ``TaskResult`` cannot be modified.
+
+.. attribute:: TaskResult.task
+
+    The :class:`Task` the result was enqueued for.
+
+.. attribute:: TaskResult.id
+
+    A unique identifier for the result, which can be passed to
+    :meth:`Task.get_result`.
+
+    The format of a Task result id will depend on the backend being used.
+    Some may use numbers, others UUIDs.
+    Task ids are always strings less than 64 characters.
+
+.. attribute:: TaskResult.status
+
+    The :class:`status <ResultStatus>` of the result.
+
+.. attribute:: TaskResult.enqueued_at
+
+    The time when the Task was enqueued.
+
+    If :attr:`Task.enqueue_on_commit` was set, this is the time the transaction
+    completed.
+
+.. attribute:: TaskResult.started_at
+
+    The time when the Task began execution.
+
+.. attribute:: TaskResult.finished_at
+
+    The time when the Task finished execution, whether it failed or succeeded.
+
+.. attribute:: TaskResult.backend
+
+    The backend the result is from.
+
+.. attribute:: TaskResult.exception_class
+
+    The exception class raised when executing the Task.
+
+    If the Task has not finished, ``ValueError`` is raised. If the Task finished
+    successfully, the exception class is ``None``.
+
+.. attribute:: TaskResult.traceback
+
+    The exception traceback from the raised exception when the Task failed.
+
+    If the Task has not finished, ``ValueError`` is raised. If the Task finished
+    successfully, the traceback is ``None``.
+
+.. attribute:: TaskResult.return_value
+
+    The return value from the Task function.
+
+    If the Task did not finish successfully, :exc:`ValueError` is raised.
+
+.. method:: TaskResult.refresh
+
+    Refresh the result's attributes from the queue store.
+
+.. method:: TaskResult.arefresh
+
+    The ``async`` variant of :meth:`TaskResult.refresh`.
+
+.. attribute:: TaskResult.is_finished
+
+    Whether the Task has finished (successfully or not).
+
+Backends
+========
+
+Base backend
+------------
+
+.. module:: django.tasks.backends.base
+
+.. class:: BaseTaskBackend
+
+``BaseTaskBackend`` is the parent class for all Task backends.
+
+.. method:: BaseTaskBackend.get_result(result_id)
+
+    Retrieve a result by its id. If the result does not exist,
+    :exc:`ResultDoesNotExist <django.tasks.exceptions.ResultDoesNotExist>`
+    is raised.
+
+    If the backend does not support ``get_result``, :exc:`NotImplementedError`
+    is raised.
+
+.. method:: BaseTaskBackend.aget_result(result_id)
+
+    The ``async`` variant of :meth:`BaseTaskBackend.get_result`.
+
+Features
+~~~~~~~~
+
+Some backends may not support all features Django provides. It's possible to
+identifying the supported functionality of a backend, and potentially
+changing behavior accordingly.
```

Comment:
```suggestion
Feature flags
~~~~~~~~~~~~~

Some backends may not support all features Django provides. It's possible to
identify the supported functionality of a backend, and change behavior
accordingly.
```

Reply:
I think I would move this section to be below "Available backends"

Reply:
Sphinx isn't really happy with this. The title needs to be under the `django.tasks.backends.base` module.

 04516776dec37e42992f694c3c9edbdc5a19185c reverse the change. I'm not super familiar with Sphinx, so help appreciated!

Reply:
I think we can leave it but I am wondering if we should have things like `.. attribute:: ImmediateBackend.supports_get_result`. It's duplicate information but it means if things are complex, the need for footnotes is reduced and it's cleaner for things like `.. versionchanged` notes in future :thinking: 
I'm not sure